providers/provider.go

```diff
@@ -52,20 +58,34 @@ type SnapshotInfo struct {
 	Tags       map[string]string
 }
 
-type InstanceParameters struct {
-	MaintenanceWindow string `json:"maintenance_window"`
-	DailyBackupWindow string `json:"daily_backup_window"`
+type CacheParameter struct {
+	ParameterName  string `json:"parameter_name"`
+	ParameterValue string `json:"parameter_value"`
+}
+type ServiceParameters struct {
+	PreferredMaintenanceWindow string           `json:"preferred_maintenance_window"`
+	DailyBackupWindow          string           `json:"daily_backup_window"`
+	CacheParameters            []CacheParameter `json:"cache_parameters"`
```

Comment:
So wait do we just return a great big list of cache parameters to the user which they can't adjust?

And for the one that we do allow adjustment, we accept it being changed as a top-level key `maxmemory_policy`, but then we _return_ it as a member of a full list of cache parameters in "name" "value" pairs?

Reply:
In my opinion, these are all things which a tenant might care about. If we are exposing instance information, it seemed sensible to expose all of it. Happy to be corrected on that?

Would it suffice to add an extra 'ServiceParameter' `maxmemory_policy` and replicate the value?

I note this is also returning `daily_backup_window` - should this also be removed?

I would love to update this to have the value be set via `CacheParameters`, but that's going to change the API. I suppose we could also pull the update parameter from `cache_parameters`, and mark the top-level key as deprecated?

Thoughts?

Reply:
I'm really not sure where to go from here. Would appreciate input from other @alphagov/team-government-paas-people members

Reply:
I think just add a `maxmemory_policy` key.

I'm personally wound up by the "list of name/value" format of the full cache parameters because of how awkward they are for users to work with and the uncertainty they introduce over order-sensitivity but that's just me being me.